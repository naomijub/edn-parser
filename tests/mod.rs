#![allow(missing_docs)]

use edn_parser::*;
use insta::assert_snapshot;

mod error_cases;

#[test]
fn int() {
    assert_snapshot!(edn_parse("123").unwrap().cst);
}

#[test]
fn neg_int() {
    assert_snapshot!(edn_parse("-123").unwrap().cst);
}

#[test]
fn float() {
    assert_snapshot!(edn_parse("12.3").unwrap().cst);
}

#[test]
fn neg_float() {
    assert_snapshot!(edn_parse("-12.3").unwrap().cst);
}

#[test]
fn bool_true() {
    assert_snapshot!(edn_parse("true").unwrap().cst);
}

#[test]
fn bool_false() {
    assert_snapshot!(edn_parse("false").unwrap().cst);
}

#[test]
fn nil() {
    assert_snapshot!(edn_parse("nil").unwrap().cst);
}

#[test]
fn simple_char() {
    assert_snapshot!(edn_parse("\\c").unwrap().cst);
}

#[test]
fn newline_char() {
    assert_snapshot!(edn_parse("\\newline").unwrap().cst);
}

#[test]
fn string() {
    assert_snapshot!(edn_parse("\"hello world\"").unwrap().cst);
}

#[test]
fn list_random_types() {
    assert_snapshot!(edn_parse("(1 nil true -4.2 \"hello world\")").unwrap().cst);
}

#[test]
fn set_random_types() {
    assert_snapshot!(edn_parse("#{1 nil true -4.2 \"hello world\"}").unwrap().cst);
}

#[test]
fn vector_random_types() {
    assert_snapshot!(edn_parse("[1 nil true -4.2 \"hello world\"]").unwrap().cst);
}

#[test]
fn map_random_types() {
    assert_snapshot!(
        edn_parse("{1 nil true -4.2 \"hello world\" :value}")
            .unwrap()
            .cst
    );
}
