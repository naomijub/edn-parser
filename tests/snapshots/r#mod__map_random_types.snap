---
source: tests/mod.rs
expression: "edn_parse(\"{1 nil true -4.2 \\\"hello world\\\" :value}\").unwrap().cst"
---
edn [0..38]
    map [0..38]
        LBrace "{" [0..1]
        member [1..6]
            literal [1..2]
                Number "1" [1..2]
            Whitespace " " [2..3]
            literal [3..6]
                Null "nil" [3..6]
        Whitespace " " [6..7]
        member [7..16]
            literal [7..11]
                True "true" [7..11]
            Whitespace " " [11..12]
            literal [12..16]
                Number "-4.2" [12..16]
        Whitespace " " [16..17]
        member [17..37]
            literal [17..30]
                String "\"hello world\"" [17..30]
            Whitespace " " [30..31]
            literal [31..37]
                keyword [31..37]
                    Colon ":" [31..32]
                    Name "value" [32..37]
        RBrace "}" [37..38]
